src/Graphics/Image/Processing/Deconvolution.hs

```diff
@@ -0,0 +1,49 @@
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE BangPatterns #-} 
+
+module Graphics.Image.Processing.Deconvolution where
+
+import Graphics.Image as I
+import Graphics.Image.Interface as I
+import Prelude as P
+
+-- | Supplementary function needed to blur an image and test the algorithm.
+randomBlur :: Image VS X Double
+randomBlur = randomBlur' / scalar (I.sum randomBlur')
+  where randomBlur' = fromLists [[0, 0, 4, 3, 2]
+                                ,[0, 1, 3, 4, 3]
+                                ,[1, 2, 3, 3, 4]
+                                ,[0, 1, 2, 1, 0]
+                                ,[0, 0, 1, 0, 0]]
+
+-- | 'deconvolve' is used to correct the systematic error of blur (loss of 
+-- contrast in smaller features) in images. This particular variant is used
+-- when no information about the distortion (blurring and noise) is known.
+-- More info about deconvolution at <https://en.wikipedia.org/wiki/Deconvolution>
+-- 
+-- <<images/frog.jpg>>   <<images/frog_deconvolve.jpg>>
+-- 
+--  Usage:
+-- >>> u0 <- readImage' "images/frog.jpg" :: IO (Image VS RGB Double)
+-- >>> let p = randomBlur
+-- >>> let d = convolve Edge p u0
+-- >>> let us = iterate (deconvolve p d) d  -- Iterative deconvolution
+-- >>>     u1 = us !! 1 -- one iteration
+-- >>>     u2 = us !! 20 -- twenty iterations
+-- >>> let output = makeImage (rows u0, cols u0 * 4)
```

Comment:
This is an extremely inefficient way to concat images together, try using `leftToRight` function

Reply:
@lehins Sure sir, sorry I was unaware of that. Will change it.

Reply:
@lehins Sir, I tested the code using `leftToRight` and I think that the computation time is more or less same as compared when using the previous technique. Please let me know if you would like me to replace the current version. I'm very sorry for the delay in addressing the PR's, was a bit engaged in my academic activities. I will now take up the PR's one by one and will try to address all the issues as soon as possible.

Reply:
> the computation time is more or less same

Without some numbers/benchmarks I will not believe that. 

You probably aren't noticing the difference because deconvolution is an expensive operation and 20 iterations take long time. The operation of concatenation of 4 images the way you have it implemented in the example on its own will be much slower than if you use `leftToRight`, so I do still recommend you change it.

No worries on the delay. There isn't a particular rush on any of this, especially considering that GSoC is over. Congrats on a successful project. ;)

Reply:
@lehins I think I might be doing something wrong. I implemented it like : 
      let output1 = leftToRight u0 d
      let output2 = leftToRight output1 u1  
      let output3 = leftToRight output2 u2	
Okay sir, I'll try to give you the compile time for both cases. Thank you very much sir!!

Reply:
You are doing it right.
It's not the compile time that's important, but the runtime :)
If you'd like to benchmark the difference, you'll have to isolate that operation of concatenation from deconvolution. Best way would be to simply read one image and try to concatenate 4 copies of it together using both approaches and record the time.

Reply:
@lehins Very sorry, I actually did mean runtime but wrote compile time instead. Okay, I'll try to record the time.